src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1193,7 +1193,9 @@ private BoundExpression SynthesizeMethodGroupReceiver(CSharpSyntaxNode syntax, A
             HashSet<DiagnosticInfo> unused = null;
             if (currentType.IsEqualToOrDerivedFrom(declaringType, ignoreDynamic: false, useSiteDiagnostics: ref unused))
             {
-                return ThisReference(syntax, currentType, wasCompilerGenerated: true);
+                // PROTOTYPE: Get the type of `this` in a better way. Probably want to modify the above if statement?
+                var typeOfThis = currentType.ExtensionClassType ?? currentType;
+                return ThisReference(syntax, typeOfThis, wasCompilerGenerated: true);
```

Comment:
Please add a test that requires `IsEqualToOrDerivedFrom` to use `ExtensionClassType`.